### DISCUSSION TITLE
Tfx

### DISCUSSION THREAD

**Discussion by Orensjanelle** [iptv-org/community#929] [Q&A]
### Channel ID

TFX.fr

### Notes (optional)

I can not find this Channel with the link 

### Please confirm the following

- [X] I have read [Contributing Guide](https://github.com/iptv-org/iptv/blob/master/CONTRIBUTING.md#request-a-channel)

**Comment by ghost** [comment]
how do we get those "constant" TF1's link? (ex: https://tf1-hls-live-cdn1.tf1.fr/out/v1/c2e382be3aa2486e8753747e7bb6157e/index.m3u8)

**Comment by Sphinxroot** [comment]
original TFX TF1S TMC are now tokenized
VLC

#EXTINF:-1 ,TF1S
#EXTVLCOPT:http-referrer=http://www.callofliberty.fr/tv/20-TF1SERIESFILMS/20-TF1SERIESFILMS.php
http://www.callofliberty.fr/stream/TF1SERIESFILMS/master.m3u8
#EXTINF:-1 ,TMC
#EXTVLCOPT:http-referrer=http://www.callofliberty.fr/tv/10-TMC/10-TMC.php
http://www.callofliberty.fr/stream/TMC/master.m3u8
#EXTINF:-1 ,TFX
#EXTVLCOPT:http-referrer=http://www.callofliberty.fr/tv/11-TFX/11-TFX.php
http://www.callofliberty.fr/stream/TFX/master.m3u8

kodi
#EXTINF:-1 ,TF1S
http://www.callofliberty.fr/stream/TF1SERIESFILMS/master.m3u8|Referer=http://www.callofliberty.fr/tv/20-TF1SERIESFILMS/20-TF1SERIESFILMS.php
#EXTINF:-1 ,TMC
http://www.callofliberty.fr/stream/TMC/master.m3u8|Referer=http://www.callofliberty.fr/tv/10-TMC/10-TMC.php
#EXTINF:-1 ,TFX
http://www.callofliberty.fr/stream/TFX/master.m3u8|Referer=http://www.callofliberty.fr/tv/11-TFX/11-TFX.php

alt
#EXTINF:0,3. TF1
http://livetv.ktv.zone/3/play.m3u8
#EXTINF:0,22. TF1 Séries Films
http://livetv.ktv.zone/22/play.m3u8
#EXTINF:0,12. TMC
http://livetv.ktv.zone/12/play.m3u8
#EXTINF:0,13. TFX
http://livetv.ktv.zone/13/play.m3u8

**Comment by BellezaEmporium** [comment]
> how do we get those "constant" TF1's link? (ex: https://tf1-hls-live-cdn1.tf1.fr/out/v1/c2e382be3aa2486e8753747e7bb6157e/index.m3u8)

It would need a POST to https://compte.tf1.fr/accounts.login

loginID= *email* &password= *password* &sessionExpiration=31536000&targetEnv=jssdk&include=identities-all%2Cdata%2Cprofile%2Cpreferences%2C&includeUserInfo=true&loginMode=standard&lang=fr&APIKey=3_hWgJdARhz_7l1oOp3a8BDLoR9cuWZpUaKG4aqF7gum9_iK3uTZ2VlDBl8ANf8FVk&sdk=js_latest&authMode=cookie&pageURL=https%3A%2F%2Fwww.tf1.fr%2F&sdkBuild=13987&format=json

You then grab the UIDSignature and the UID from the Gigya response (either from the accounts login, either from the userInfo, UID + UIDSignature).

You then POST to https://www.tf1.fr/token/gigya/web, in order for this request to work you will also need to create a timestamp in EPOCH/UNIX format (something like 1687780538)

{"uid":" *UID* ","signature":" *UIDSignature* ","timestamp":,"consent_ids":["1","2","3","4","10001","10003","10005","10007","10013","10015","10017","10019","10009","10011","13002","13001","10004","10014","10016","10018","10020","10010","10012","10006","10008"]

You grab the "token" from the JSON answer.

After that, you can do a GET request to https://mediainfo.tf1.fr/mediainfocombo/L_TF1?pver=5010000&context=MYTF1&topDomain=unknown&platform=web&device=desktop&os=windows&osVersion=10.0&playerVersion=5.10.0&productName=mytf1&productVersion=2.59.2&browser=firefox&browserVersion=114 with a specific header Authorization : Bearer token.

Answer will be in "delivery" -> "url"

**Comment by ghost** [comment]
by the way, we don't have to create a timestamp, it is referenced in the "signatureTimestamp" property!

**Comment by ghost** [comment]
but the thing is that i'm talking about not-tokenized url

**Comment by adrian-squared** [comment]
hey, could someone explain to me how to do all this POST stuff please ? i've been able to access a URL with what seems like a token using the network tab on firefox, but it doen't seem to work. It could just be my internet though.

**Comment by BellezaEmporium** [comment]
> but the thing is that i'm talking about not-tokenized url

Doesn't exist anymore, officially.

**Comment by adrian-squared** [comment]
after checking, i do have a URL with a token that works, but it stops working after a bit. i was able to get it to work fully, but i don't know how

**Comment by adrian-squared** [comment]
and right after i post that it started working again, should i post the link ? i know it's got my token but my account is literally useless so whatever

**Comment by BellezaEmporium** [comment]
> and right after i post that it started working again, should i post the link ? i know it's got my token but my account is literally useless so whatever

The token changes over a few hours, so I don't think it would work.

**Comment by freearhey** [comment]
https://github.com/iptv-org/iptv/blob/aec1bdb0a7037bf6f7112d2be3bdd93a50d5209f/streams/fr.m3u#L228-L229
